Mage.Sets/src/mage/cards/s/SoldeviSentry.java

```diff
@@ -45,37 +48,69 @@ public SoldeviSentry copy() {
     }
 }
 
-class SoldeviSentryEffect extends RegenerateSourceEffect {
+class SoldeviSentryRegenerateEffect extends RegenerateSourceEffect {
 
-    public SoldeviSentryEffect() {
+    public SoldeviSentryRegenerateEffect() {
         super();
         this.staticText = "Choose target opponent. Regenerate {this}. When it regenerates this way, that player may draw a card";
     }
 
-    protected SoldeviSentryEffect(final SoldeviSentryEffect effect) {
+    protected SoldeviSentryRegenerateEffect(final SoldeviSentryRegenerateEffect effect) {
         super(effect);
     }
 
     @Override
-    public SoldeviSentryEffect copy() {
-        return new SoldeviSentryEffect(this);
+    public SoldeviSentryRegenerateEffect copy() {
+        return new SoldeviSentryRegenerateEffect(this);
     }
 
     @Override
     public boolean apply(Game game, Ability source) {
         //20110204 - 701.11
-        Player opponent = game.getPlayer(source.getFirstTarget());
         Permanent permanent = game.getPermanent(source.getSourceId());
         if (permanent != null && permanent.regenerate(source, game)) {
-            if (opponent != null) {
-                if (opponent.chooseUse(Outcome.DrawCard, "Draw a card?", source, game)) {
-                    opponent.drawCards(1, source, game);
-                }
-            }
+            game.fireReflexiveTriggeredAbility(
+                    new ReflexiveTriggeredAbility(
+                            new SoldeviSentryEffect().setTargetPointer(
+                                    new FixedTarget(targetPointer.getFirst(game, source), game)
+                            ), false
+                    ).setTriggerPhrase("When it regenerates this way, "),
+                    source
+            );
             this.used = true;
             return true;
         }
         return false;
     }
-
 }
+
+class SoldeviSentryEffect extends OneShotEffect {
```

Comment:
is there a reason not to use `new DrawCardTargetEffect(1, true)` ?

Reply:
Thanks, i'll use that.
I too quickly searched for "target player may draw", did not find any match, and assumed the common effect would not allow for optional draw of non-source-controller player.